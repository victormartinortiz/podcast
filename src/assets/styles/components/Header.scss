header {
  color: #fff;
  background: #8756ca;
  padding: 15px;
  text-align: center;
    & a{
      text-decoration: none;
    }
}
