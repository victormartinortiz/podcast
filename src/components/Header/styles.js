/* eslint-disable import/prefer-default-export */
import styled from 'styled-components';

export const HeaderStyle = styled.header`
  color: #fff;
  background: #8756ca;
  padding: 15px;
  text-align: center;
  font-weight: bold;
`;

export const HeaderLink = styled.a`
  text-decoration: none;
  cursor: pointer;
`;
