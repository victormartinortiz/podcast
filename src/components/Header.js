import React from 'react';
import Link from 'next/link';
import '../assets/styles/components/Header.scss'

const Header = ({ title }) => (
  <header>
    <Link href='/'>
      <a>{title}</a>
    </Link>
  </header>
);

export default Header;
