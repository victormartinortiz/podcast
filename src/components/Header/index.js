import React from 'react';
import Link from 'next/link';
import { HeaderStyle, HeaderLink } from './styles';

const Header = ({ pageHeader }) => (
  <HeaderStyle>
    <Link href='/'>
      <HeaderLink>{pageHeader}</HeaderLink>
    </Link>
  </HeaderStyle>
);

export default Header;
