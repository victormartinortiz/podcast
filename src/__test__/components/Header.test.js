import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';

describe('<Header />', () => {
  it('render header unchanged', () => {
    const tree = create(<Header />);
    expect(tree.toJSON()).toMatchSnapshot()
  });

  it('renders Header component', () => {
    const header = shallow(<Header />);
    expect(header.length).toEqual(1);
  });

  it('renders Header title', () => {
    const header = mount(<Header />);
    expect(header.find('a').text()).toEqual('');
  });
});
