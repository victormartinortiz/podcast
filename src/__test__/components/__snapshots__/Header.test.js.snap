// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render header unchanged 1`] = `
<header
  className="styles__HeaderStyle-sc-1iihxp8-0 iWPpDd"
>
  <a
    className="styles__HeaderLink-sc-1iihxp8-1 QAMUQ"
    onClick={[Function]}
    onMouseEnter={[Function]}
  />
</header>
`;
